src/Shell.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Shell.h"
+
+void Shell::displayPrompt() {
+    cout << userNameColor << User::getCurrentUser() << resetColor
+         << " " << directoryColor << Directory::getCurrentDirectory() << resetColor;
+    m_branch = Git::getCurrentBranch();
+
+    // If the current directory is a git repository, display the current m_branch
+    if (!m_branch.empty()) {
+        cout << " " << branchColor << "(" << m_branch << ")" << resetColor;
+    }
+
+    cout << " >> ('exit' to quit): ";
+}
+
+
+void Shell::displayJobs() {
+    auto bgProcesses = MyJobs::getJobs();
+    cout << "PID\t\tCommand\t\t\tStatus" << endl;
+    for (auto it = bgProcesses.begin(); it != bgProcesses.end(); ++it) {
+        int status;
+        int processStatus = waitpid(it->getPid(), &status, WNOHANG);
+        if (processStatus == -1) {
+            std::cout << "Error checking process status for PID: " << it->getPid() << std::endl;
+        } else if (processStatus == it->getPid()) {
+            MyJobs::eraseJob(int(it - bgProcesses.begin()));
```

Comment:
This is unsafe. You are deleting the element from your map. What happens to the iterator pointing at it `it`? Can you be sure that it's still valid and can be incremented?